### ISSUE TITLE
socket headers nil

### ISSUE THREAD

**Issue by mrloop** [faye/faye-websocket-ruby#82]
I consistently get an error 
```
NoMethodError: undefined method `[]' for nil:NilClass
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/faye-1.1.2/lib/faye/transport/web_socket.rb:76:in `block in connect'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/api/event_target.rb:44:in `call'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/api/event_target.rb:44:in `dispatch_event'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/api.rb:99:in `open'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/api.rb:39:in `block in initialize'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver/event_emitter.rb:39:in `call'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver/event_emitter.rb:39:in `block in emit'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver/event_emitter.rb:38:in `each'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver/event_emitter.rb:38:in `emit'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver.rb:131:in `open'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver/client.rb:72:in `parse'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/api.rb:145:in `parse'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/client.rb:75:in `receive_data'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/eventmachine-1.0.8/lib/eventmachine.rb:193:in `run_machine'
        from /home/ewan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/eventmachine-1.0.8/lib/eventmachine.rb:193:in `run'
```
Commenting out https://github.com/faye/faye/blob/28b2f593df0897b2a18d128c566c4bc415b41a87/lib/faye/transport/web_socket.rb#L75 and I publish to faye as expected. 
Or this also works for me https://github.com/mrloop/faye-websocket-ruby/commit/fbf34e98021e09f73138643f72babdcb6749920c

**Comment by lamielle** [comment]
+1, we just updated to the latest faye-websocket and websocket-driver gems (versions 0.6.3 and 0.10.1 respectively) and we're seeing what I believe is the same crash:

    <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/faye-1.1.2/lib/faye/transport/web_socket.rb:75:in `block in connect': undefined method `[]' for nil:NilClass (NoMethodError)
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/api/event_target.rb:44:in `call'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/api/event_target.rb:44:in `dispatch_event'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/api.rb:99:in `open'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/api.rb:39:in `block in initialize'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver/event_emitter.rb:39:in `call'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver/event_emitter.rb:39:in `block in emit'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver/event_emitter.rb:38:in `each'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver/event_emitter.rb:38:in `emit'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver.rb:131:in `open'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/websocket-driver-0.6.3/lib/websocket/driver/client.rb:72:in `parse'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/api.rb:145:in `parse'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/faye-websocket-0.10.1/lib/faye/websocket/client.rb:75:in `receive_data'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/eventmachine-1.0.8/lib/eventmachine.rb:193:in `run_machine'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/eventmachine-1.0.8/lib/eventmachine.rb:193:in `run'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/backends/base.rb:73:in `start'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/server.rb:162:in `start'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/controllers/controller.rb:87:in `start'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/runner.rb:200:in `run_command'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/runner.rb:156:in `run!'
	from <user>/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thin-1.6.4/bin/thin:6:in `<top (required)>'
	from <user>/.rbenv/versions/2.2.2/bin/thin:23:in `load'
	from <user>/.rbenv/versions/2.2.2/bin/thin:23:in `<top (required)>'
	from -e:1:in `load'
	from -e:1:in `<main>'

The relevant block is:

      socket.onopen = lambda do |*args|
        store_cookies(socket.headers['Set-Cookie'])
        @socket = socket
        @state = CONNECTED
        @ever_connected = true
        ping
        set_deferred_status(:succeeded, socket)
      end

**Comment by jcoglan** [comment]
This is fixed by #83.
